docs/pages/project/contributing/contributing-error.md

```diff
@@ -9,27 +9,139 @@ category: contributing
+  This tool extracts this information from the code and exports it.
+
+- By convention, error codes and the factory functions live in files called error.go. The tool checks all files, but updates only error.go files.
 
-Use the `errors.New()` function to create a new instance of the error object and pass situation-specific attributes as function arguments. See the following example for reference.
+- This tool will create a couple of files, one of them is designed to be used to generate the error reference on the meshery website.
+  The file errorutil_analyze_summary.json contains a summary of the analysis, notably lists of duplicates etc.
+
+- The tool requires a file called component_info.json. Its location can be customized, by default it is the root directory (-d flag). 
+  This file has the following content, with concrete values specific for each component:
+  {
+    "name": "meshkit",
+    "type": "library",
+    "next_error_code": 11010
+  }
+
+- The tool updates next_error_code.
+
+
+## Example
+
+In this example we are trying to create an Error for being unable to marshal JSON
 
 ```code
 var (
-    ErrExampleCode = "111"
+    // Error code 
+    ErrMarshalCode= "replace_me"
 
-    // Static errors
+    //Static errors (for example)
     ErrExample = errors.New(ErrExampleCode, errors.Alert, []string{"<short-description>"}, []string{"<long-description>"}, []string{"<probable-cause>"}, []string{"<suggested remediation>"})
 )
 
 // Dynamic errors
-func ErrExample(err error) error {
-    return errors.New(ErrExampleCode, errors.Alert, []string{"<short-description>"}, []string{"<long-description>"}, []string{"<probable-cause>"}, []string{"<suggested remediation>"})
+//Error Name
+func ErrMarshal(err error, obj string) error {
+	return errors.New(ErrMarshalCode, errors.Alert, []string{"Unable to marshal the : ", obj}, []string{err.Error()}, []string{}, []string{})
 }
+
 ```
+
+### Replacing old Error Codes 
+
+ Old
+ ```Code 
+    bd, err := json.Marshal(providers)
+	if err != nil {
+		http.Error(w, "unable to marshal the providers", http.StatusInternalServerError)
+		return
+	}
+ ```
+New
+  ```Code 
+    bd, err := json.Marshal(providers)
+    if err != nil {
+            obj := "provider"
+            http.Error(w, ErrMarshal(err, obj).Error(), http.StatusInternalServerError)
+            return
+        }
+ ```
+
+ 
+## Replacing logrus 
+ There already exists an interface for logger in <a href="https://github.com/layer5io/meshkit/blob/master/logger/logger.go">Meshkit </a>  
+
+#### Defining a Logger 
+
+   ```Code 
+    type Logger struct {
+        log   logger.Handler
+    }
+ ```
+
+#### Debug
+ 
+ Old
+
+    `logrus.Debugf("meshLocationURL: %s", meshLocationURL)`
+ New
+
+    `l.log.Debug("meshLocationURL: ", meshLocationURL)`
+
+#### Error
+
+##### Old
+  `logrus.Errorf("error marshaling data: %v.", err)`
+##### New
+  `l.log.Error(ErrMarshal(err, obj))`
+
```

Comment:
We can also add the `logrus.Infof` to `l.log.Info`

Reply:
i havnt used it yet, could write the syntax for it